packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts

```diff
@@ -162,13 +178,20 @@ export const sqlRunnerSlice = createSlice({
                     payload: state.sql,
                     type: 'sql',
                 });
+                state.hasUnrunChanges = false;
             }
         },
         updateName: (state, action: PayloadAction<string>) => {
             state.name = action.payload;
         },
         setSql: (state, action: PayloadAction<string>) => {
             state.sql = action.payload;
+
+            const normalizedNewSql = normalizeSQL(action.payload);
+            const normalizedCurrentSql = normalizeSQL(
+                state.successfulSqlQueries.current || '',
+            );
+            state.hasUnrunChanges = normalizedNewSql !== normalizedCurrentSql;
```

Comment:
this was mentioned by you @rephus on another PR, this uses `sql-formatter` as the comparison tool between previous and current sql queries. If there's extra whitespace, then it shouldn't consider them as different. Updated the history adapter to accept a comparison function when removing duplicates.